homeassistant/components/weatherflow_cloud/__init__.py

```diff
@@ -2,33 +2,126 @@
 
 from __future__ import annotations
 
+import asyncio
+from dataclasses import dataclass
+
+from weatherflow4py.api import WeatherFlowRestAPI
+from weatherflow4py.models.rest.stations import StationsResponseREST
+from weatherflow4py.models.ws.obs import WebsocketObservation
+from weatherflow4py.models.ws.types import EventType
+from weatherflow4py.models.ws.websocket_request import (
+    ListenStartMessage,
+    RapidWindListenStartMessage,
+)
+from weatherflow4py.models.ws.websocket_response import EventDataRapidWind, RapidWindWS
+from weatherflow4py.ws import WeatherFlowWebsocketAPI
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_API_TOKEN, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DOMAIN
-from .coordinator import WeatherFlowCloudDataUpdateCoordinator
+from .const import DOMAIN, LOGGER
+from .coordinator import (
+    WeatherFlowCloudDataCallbackCoordinator,
+    WeatherFlowCloudUpdateCoordinatorREST,
+)
 
 PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.WEATHER]
 
 
+@dataclass
+class WeatherFlowCoordinators:
+    """Data Class for Entry Data."""
+
+    rest: WeatherFlowCloudUpdateCoordinatorREST
+    wind: WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, RapidWindListenStartMessage, RapidWindWS
+    ]
+    observation: WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, ListenStartMessage, WebsocketObservation
+    ]
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up WeatherFlowCloud from a config entry."""
 
-    data_coordinator = WeatherFlowCloudDataUpdateCoordinator(
+    LOGGER.debug("Initializing WeatherFlowCloudDataUpdateCoordinatorREST coordinator")
+
+    rest_api = WeatherFlowRestAPI(
+        api_token=entry.data[CONF_API_TOKEN], session=async_get_clientsession(hass)
+    )
+
+    stations: StationsResponseREST = await rest_api.async_get_stations()
```

Comment:
```suggestion
    stations = await rest_api.async_get_stations()
```
Type overwrite should not be needed